racetrack_client/racetrack_client/main.py

```diff
@@ -63,9 +64,23 @@ def _deploy(
     extra_vars: Optional[List[str]] = typer.Option(None, '-e', '--extra-vars', help='key=value pairs overriding manifest values'),
 ):
     """Send request deploying a Job to the Racetrack cluster"""
+    # Catch if something was piped to `deploy`. Works best when the build_context is git
+    # Since most of send_deploy_request assumes a directory and a manifest file, we create one from the piped data
+    piped_data = ""
+    file_path = ""
+    if not sys.stdin.isatty() and (piped_data := sys.stdin.read()):
```

Comment:
Does `sys.stdin.read()` blocks forever if the stdin buffer is empty?

Reply:
I think it is supposed to.

We can use a selector with a timeout of 0.5 seconds or something.

Reply:
According to https://stackoverflow.com/questions/3762881/how-do-i-check-if-stdin-has-some-data, maybe that's enough to just check `not sys.stdin.isatty()`

Reply:
I'd prefer having just the `not sys.stdin.isatty()` check, instead of the whole selector thing. The call will still block if something is piping very slowly, but that seems like a different problem.

Reply:
Okay, that sounds good to me.